include/ginkgo/core/base/executor.hpp

```diff
@@ -1143,26 +1213,61 @@ class CudaExecutor : public detail::ExecutorBase<CudaExecutor>,
         return cusparse_handle_.get();
     }
 
+    /**
+     * Get the closest CPUs
+     *
+     * @return  the array of CPUs closest to this device
+     */
+    std::vector<int> get_closest_cpus() const
+    {
+        return this->cuda_exec_info_.closest_cpu_ids;
+    }
+
+    /**
+     * Gets the exec info struct
+     *
+     * @return  the exec_info struct
+     */
+    cuda_exec_info get_exec_info() const { return this->cuda_exec_info_; }
+
+    /**
+     * Bind the current executor thread to the list of cpus
+     *
+     * @param  ids  the logical ids of the cpus.
+     * @param  num_ids  the number of cpus to bind
+     */
+    void bind_to_cpus(const int *ids, const size_type num_ids) const;
+
+    /**
+     * Bind the current executor thread to the list of cores
+     *
+     * @param  ids  the logical ids of the cores.
+     * @param  num_ids  the number of cores to bind
+     */
+    void bind_to_cores(const int *ids, const size_type num_ids) const;
```

Comment:
This seems inconsistent: You return `std::vector<int>` at `get_closest_cpus()`, but here, you want to get pointers and the size separately (so you can't do `bind_to_cores(get_closest_cpus())`)?
Please use `std::vector` here as well (if you want support for arrays, add another function taking `std::array<int, N>` as an argument).

Reply:
Good point. I changed everything to take vector now and an additional function for a scalar.